features/flows/classification/engine/impl/src/main/java/org/opennms/netmgt/flows/classification/internal/DefaultClassificationService.java

```diff
@@ -79,11 +87,14 @@ public DefaultClassificationService(ClassificationRuleDao classificationRuleDao,
                                         SessionUtils sessionUtils) {
         this.classificationRuleDao = Objects.requireNonNull(classificationRuleDao);
         this.classificationGroupDao = Objects.requireNonNull(classificationGroupDao);
-        this.classificationEngine = Objects.requireNonNull(classificationEngine);
+        this.classificationEngine = new AsyncReloadingClassificationEngine(Objects.requireNonNull(classificationEngine));
         this.sessionUtils = Objects.requireNonNull(sessionUtils);
         this.ruleValidator = new RuleValidator(filterService);
         this.groupValidator = new GroupValidator(classificationRuleDao);
         this.csvService = new CsvServiceImpl(ruleValidator);
+        // trigger reload
+        // -> blocks classification requests until classification engine is ready
+        this.classificationEngine.reload();
```

Comment:
Initial reload should be done by `AsyncReloadingClassificationEngine` itself. This would make the wrapper more encapsulated.

Reply:
done